Neos.Neos/Classes/Utility/NodeUriPathSegmentGenerator.php

```diff
@@ -38,19 +41,36 @@ class NodeUriPathSegmentGenerator
      */
     public function generateUriPathSegment(?Node $node = null, ?string $text = null): string
     {
-        $language = null;
-        if ($node) {
-            $text = $text ?: $node->getLabel() ?: ($node->nodeName?->value ?? '');
-            $languageDimensionValue = $node->originDimensionSpacePoint->coordinates['language'] ?? null;
-            if (!is_null($languageDimensionValue)) {
-                $locale = new Locale($languageDimensionValue);
-                $language = $locale->getLanguage();
-            }
-        } elseif (is_null($text) || empty($text)) {
+        if ($node === null && empty($text)) {
             throw new Exception('Given text was empty.', 1457591815);
         }
-        $text = $this->transliterationService->transliterate($text, $language);
 
-        return Transliterator::urlize($text);
+        $textForNode = $text ?: $node->getLabel() ?: $node->nodeName?->value ?? '';
+
+        return $this->generateUriPathSegmentFromTextForDimension(
+            $textForNode,
+            $node?->originDimensionSpacePoint->toDimensionSpacePoint() ?? DimensionSpacePoint::fromArray([])
+        );
+    }
+
+    /**
+     * Generates a URI path segment for a given text taking the language dimension value into account
+     *
+     * @param string $text
+     * @param DimensionSpacePoint $dimensionSpacePoint to determine language dimension value from
+     */
+    public function generateUriPathSegmentFromTextForDimension(string $text, DimensionSpacePoint $dimensionSpacePoint): string
+    {
+        $languageDimensionValue = $dimensionSpacePoint->getCoordinate(new ContentDimensionId('language'));
```

Comment:
we should avoid to hardcode `language` as its only currently the best possible guess. But there is nothing wrong with naming your language dimension `lang`

This configuration should live in Neos.Neos and might look like

```yaml
Neos:
  Neos:
    # Used in places to guess the correct language from a DimensionSpacePoint
    languageDimensionKey: "language"
```

Reply:
Would be also helpful here:
https://github.com/neos/neos-development-collection/blame/9.0/Neos.Neos/Classes/View/FusionViewI18nTrait.php#L42-L43

Or in SEO Package:
https://github.com/neos/neos-seo/blob/c047aeab7b63e860cedc33c45a36a2a1a2e28490/Resources/Private/Fusion/Metadata/AlternateLanguageLinks.fusion#L7
https://github.com/neos/neos-seo/blob/c047aeab7b63e860cedc33c45a36a2a1a2e28490/Resources/Private/Fusion/RobotsTxt/RobotsTxt.fusion#L10

Reply:
configurable in Neos.Neos.sites.*.contentDimensions.languageDimensionKey

Reply:
quick comment after discussing this in our weekly: if we make this configurable, this should probably be a setting that is specific to a *content repository* but this has some implications, e.g. "locale" is a Flow specific concept currently